serde-oracle-jdbc-json/src/test/java/io/micronaut/serde/bson/SampleData.java

```diff
@@ -0,0 +1,126 @@
+package io.micronaut.serde.bson;
+
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.serde.annotation.Serdeable;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonBinarySerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonBinaryStringSerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonDurationSerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonInstantSerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonLocaleDateSerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonLocaleDateTimeSerde;
+import io.micronaut.serde.oracle.jdbc.json.serde.OracleJsonOffsetDateTimeSerde;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.Period;
+import java.util.UUID;
+
+@Introspected
+public class SampleData {
+
+    @Serdeable.Deserializable(using = OracleJsonLocaleDateTimeSerde.class)
+    @Serdeable.Serializable(using = OracleJsonLocaleDateTimeSerde.class)
+    private LocalDateTime localDateTime;
+
+    @Serdeable.Deserializable(using = OracleJsonOffsetDateTimeSerde.class)
+    @Serdeable.Serializable(using = OracleJsonOffsetDateTimeSerde.class)
+    private OffsetDateTime offsetDateTime;
+
+    @Serdeable.Deserializable(using = OracleJsonLocaleDateSerde.class)
+    @Serdeable.Serializable(using = OracleJsonLocaleDateSerde.class)
+    private LocalDate date;
+
+    @Serdeable.Deserializable(using = OracleJsonInstantSerde.class)
```

Comment:
This is how we need to configure fields if we want to read from Oracle JSON and be able to write back to. It means these beans might require DTO for data transfer over web because it uses serialization for DATA json.

Reply:
this shouldn't be necessary if we order the serdes and perhaps add a way for serde to only be active if the encoder/decoder is of a certain type

Reply:
Yes, this would not be very user friendly, but for entity that will be mapped to (Oracle) JSON we must (de)serialize field using this serde. However, for HTTP client, we would use different one and separate DTO so we might need to do it this way and document with examples.

Reply:
i think it should be opt-in like in these annotations, since not everyone may want to use the db-specific data types, and because they might lose information (e.g. precision or some tz info)

Reply:
With some back and forth, this is not needed anymore as Graeme designed serde registration that will allow us to use serdes within Oracle JSON parsing context without need to register.